dev/Saumya/algorithm_helpers.py

```diff
@@ -0,0 +1,321 @@
+
+
+        #save MLA predictions that could be used later
+        alg.fit(X, y)
+        MLA_predict[MLA_name] = alg.predict(X)
+
+        row_index += 1
+
+
+    #print and sort table
+    MLA_compare.sort_values(by = ['MLA Test Accuracy Mean'], ascending = False, inplace = True)
+    return MLA_compare, MLA_predict
+
+def run_DT_model(X, y, cv_split):
+    #base model
+    dtree = tree.DecisionTreeClassifier(random_state = 42)
+    base_results = model_selection.cross_validate(dtree, X, y, cv  = cv_split, return_train_score=True, n_jobs=-1)
+    dtree.fit(X, y)
+
+    print('BEFORE DT Parameters: ', dtree.get_params())
+    print("BEFORE DT Training accuracy: {:.2f}". format(base_results['train_score'].mean()*100)) 
+    print("BEFORE DT Test accuracy: {:.2f}". format(base_results['test_score'].mean()*100))
+    print("BEFORE DT Test score 3*std: +/- {:.2f}". format(base_results['test_score'].std()*100*3))
+    print("BEFORE DT Test accuracy (min): {:.2f}". format(base_results['test_score'].min()*100))
+    print('-'*10)
+
+
+    #tune hyper-parameters
+    param_grid = {'criterion': ['gini', 'entropy'],  #scoring methodology; two supported formulas for calculating information gain; default is gini
+                  'splitter': ['best', 'random'], #splitting methodology; two supported strategies; default is best
+                  'max_depth': [2,4,6,8,10,None], #max depth that the tree can grow; default is none
+                  'min_samples_split': [2,5,10,.03,.05], #minimum subset size BEFORE new split (fraction is % of total); default is 2
+                  'min_samples_leaf': [1,5,10,.03,.05], #minimum subset size AFTER new split split (fraction is % of total); default is 1
+                  'max_features': [None, 'auto'], #max features to consider when performing split; default is None, i.e., all
+                  'random_state': [0] #seed for random number generator, hence not required to be tuned
+                 }
+
+    #choose the best model with grid search
+    tuned_model = model_selection.GridSearchCV(tree.DecisionTreeClassifier(), param_grid=param_grid, scoring = 'accuracy', cv = cv_split, return_train_score=True, n_jobs=-1)
+    tuned_model.fit(X, y)
+
+    print('AFTER DT Parameters: ', tuned_model.best_params_)
+    print("AFTER DT Training accuracy: {:.2f}". format(tuned_model.cv_results_['mean_train_score'][tuned_model.best_index_]*100)) 
+    print("AFTER DT Test accuracy: {:.2f}". format(tuned_model.cv_results_['mean_test_score'][tuned_model.best_index_]*100))
+    print("AFTER DT Test 3*std: +/- {:.2f}". format(tuned_model.cv_results_['std_test_score'][tuned_model.best_index_]*100*3))
+    print('-'*10)
+    return tuned_model
+
+def run_voting_model(X, y, cv_split):
+    
+    #Removed models without attribute 'predict_proba' (required for vote classifier) and models with a 1.0 correlation to another model
+    #Using default hyper-parameters
+    vote_est = [
+        #Ensemble Methods
```

Comment:
Maybe turn `get_algorithms` output into a dict, and just list the keys here, so as to avoid reinitializing the models. Not really a performance issues, more code consistency and maintenance.

Reply:
That's right, the code looks a lot better now. Thank you :)